src/shogun/base/SGObject.h

```diff
@@ -76,19 +76,23 @@ template <class T> class SGStringList;
 #define VARARG_IMPL(base, count, ...) VARARG_IMPL2(base, count, __VA_ARGS__)
 #define VARARG(base, ...) VARARG_IMPL(base, VA_NARGS(__VA_ARGS__), __VA_ARGS__)
 
-#define SG_ADD4(param, name, description, ms_available) {\
-		m_parameters->add(param, name, description);\
-		if (ms_available)\
-			m_model_selection_parameters->add(param, name, description);\
-}
+#define SG_ADD4(param, name, description, ms_available)                        \
+	{                                                                          \
+		m_parameters->add(param, name, description);                           \
+		watch_param(name, param);                                              \
```

Comment:
this reminds me. We currently have two maps: one for registering all paramters, one for modelselection available only.
I suggest we don't keep this design with tags, but rather have one and somehow mark them otherwise ?

Reply:
I think the parameter we store should be 'rich'. This means it is not just 'Any' but 'Any' + extras. Such as this model_selection_enabled.

Reply:
Yep I agree. I think this is a place to include the domain bounds, information about gradients, sampling callbacks (?), priors, etc